docs/debug.md

```diff
@@ -0,0 +1,173 @@
+    AZURE_SUBSCRIPTION_ID=<YOUR_AZURE_SUBSCRIPTION_ID>
+    AZURE_TENANT_ID=<YOUR_AZURE_TENANT_ID>
+    ```
+2. Query the values for each of the keys above.
+    ```bash
+    # KEYVAULT_RESOURCE_GROUP
+    $ echo $RESOURCEGROUP
+
+    # KEYVAULT_NAME
+    $ echo $KEYVAULT_NAME
+
+    # KEYVAULT_CLIENT_ID
+    $ echo $CLIENT_ID
+
+    # KEYVAULT_CLIENT_SECRET
+    $ echo $CLIENT_SECRET
+
+    # AZURE_SUBSCRIPTION_ID
+    $ AZURE_SUBSCRIPTION_ID=$(az account show --query id -o tsv)
+    $ echo $AZURE_SUBSCRIPTION_ID
+
+    # AZURE_TENANT_ID
+    $ AZURE_TENANT_ID=$(az account show --query tenantId -o tsv)
+    $ echo $AZURE_TENANT_ID
+    ```
+
+3. Once you've added the keys to the secrets, run the following task to generate your `launch.json` file.
+    ```
+    make setup-debug
+    ```
+    - You should see a `launch.json` populate into your `debug` directory.
+
+### launch.json
+
+  - the `launch.json` file includes an `args` property which has a value of an array of strings. Each string is a command line argument that is needed to be passed to the `secrets-store-csi-driver-provider-azure` executable.
+      - `--attributes` Are the parameters that correspond to the `SecretProviderClass`
+      - `--secrets` Are secrets that are stored within the key vault.
+      - `--permission` File permissions. Set to `422`. (422 relates to 644 (Octal) `-rw-r--r--`)
+      - `--debug` set to **true** to allow for debugging of the executable
+
+### parameters.yaml
+
+  - This is a `yaml` file that maps out the attributes for the `secrets-store-csi-driver-provider-azure`.
+  - The `objects` property takes a string that contains a stringed version of an **array of objects.**
+  - **NOTE:** The `objects` property defines which keys should be present within the Key Vault. This is where you would add whichever secrets, credentials, or keys are expected to be accessed within the Azure Key Vault.
+      - Example: If a key named `anExampleKey` was present inside the Azure Key Vault. Another stringed object would be  added inside the `array:` as demonstrated below:
+          ```yaml
+          objects:  |
+            array:
+              - |
+                objectName: clientId
+                objectType: secret
+              - |
+                objectName: clientSecret
+                objectType: secret
+              - |
+                objectName: anExampleKey
+                objectType: key
+          ```
+
+### build-args
+
+- This file contains a shell script for building the `launch.json`
+- `SECRETS` env variablepulls the `AZURE_CLIENT_ID` and `AZURE_CLIENT_SECRET` from the `secrets.env` and serializes them into a string to be passed to the `--secrets` arg within the `launch.json`
+- the `ATTRIBUTES` replaces the env variables taken from `secrets.env`  into their correct places within the `paramters.yaml`. It then `cat` the file, converts it to JSON, and saves the output to the `ATTRIBUTES` env variable.
```

Comment:
Describe the **why** rather than the **how** here. If the reader wants to understand the **how**, they can dig into the shell script.